### ISSUE TITLE
Is it supposed to always redo the embeddings?

### ISSUE THREAD

**Issue by avataraustin** [brianpetro/smart-connect#21]
I just updated to the newest version of SC just to be sure everything was up to date.
It went through the usual of popping up the dialogue of "Embeddings smart_notes..."
It then did the "are you ready to begin embedding smart_blocks?" and it did start and finish that.

My question is whether this should happen every single time I open Obsidian?  Just to check I immediately closed and reopened Obsidian without changing anything at all.  It then went through the same process of "Embedding the smart_notes" again and also once again went through the process of Asking me to embed the smart blocks which I once again completed.  Just to be sure I went through this process again and closed/reopened obsidian without doing anything in Obsidian at all.  It once again went through the process of embedding smart_notes and then asking to embed smart_blocks. The numbers were even the same.
I remember when SC used to want to sometimes embed ALL my notes again when I would reopen Obsidian so admittedly this is something I feel I can live with but it seemed like something I should mention.

**Comment by brianpetro** [comment]
Hey @avataraustin 

No, it's not supposed to re-embed every time you restart Obsidian. 

There should be error logs either 

A) when loading the embeddings or

B) when saving the embeddings

If you can find some errors at either of those points then I can see what might be causing the issue.

Thanks for bringing this to my attention 
🌴

**Comment by avataraustin** [comment]
![Screenshot 2024-07-15 at 7 47 39 AM](https://github.com/user-attachments/assets/e2d95f77-0804-4db9-b4cb-980d479c1e61)
![Screenshot 2024-07-15 at 7 46 43 AM](https://github.com/user-attachments/assets/fdae8f32-1258-4332-b235-62cf9647b288)

**Comment by brianpetro** [comment]
Hey @avataraustin 

It looks like it loaded correctly that time.

Did it re-embed everything that time?

🌴

**Comment by avataraustin** [comment]
yes, it reliably starts embedding smart notes and then asks to embed smart blocks every time I open Obsidian. I wouldn't say it re-embeds _everything_ but there is definitely a certain number that it always wants to embed every time, 467 smart blocks last time.

![Screenshot 2024-07-15 at 12 50 49 PM](https://github.com/user-attachments/assets/2dce669c-65e3-4975-87ef-999bb2ebfc91)

**Comment by brianpetro** [comment]
Good to know.

Do any errors occur when embedding the blocks?

In the `.smart-env/multi/` folder you should be able to see which 86 notes are being re-embedded by sorting the files by "last modified" time. Do these recently changed files have anything in common?

Thanks for your help in solving this
🌴

**Comment by avataraustin** [comment]
I didn't notice anything. I also checked the .smart-connections/multi folder folder and when sorted by last modified they are the same order as when left alphabetical. I also did not notice anything about those notes in particular that stood out in any way. Here is the dev con pre and post running the  block embeddings.

![Screenshot 2024-07-16 at 8 28 29 AM](https://github.com/user-attachments/assets/e473a615-6a17-435b-823c-f96161a61b80)
![Screenshot 2024-07-16 at 8 29 33 AM](https://github.com/user-attachments/assets/55e86be8-4830-41a8-86ae-d4ea9776e868)
